tests/unittests/test_timeout.py

```diff
@@ -0,0 +1,294 @@
+    """
+        Test cases to verify that we backoff and retry 5 times for 'Timeout' error
+    """
+
+    def test_check_access_token(self, mocked_request, mocked_sleep):
+        """
+            Verify we backoff for Timeout error when checking access token via 'check_access_token' function
+        """
+        # mock request and raise error
+        mocked_request.side_effect = requests.Timeout
+
+        # set config
+        config = {
+            "access_token": "test_access_token",
+            "user_agent": "test_user_agent"
+        }
+
+        # verify that we raise Timeout error when using "with" statement
+        with self.assertRaises(requests.Timeout):
+            # create client and call function
+            client = TikTokClient(config.get("access_token"), config.get("user_agent"), config.get("request_timeout"))
+            client.__enter__()
+
+        # verify that we backoff for 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+
+    def test_request__check_access_token(self, mocked_request, mocked_sleep):
+        """
+            Verify we backoff for Timeout error when 'check_access_token' is called from 'request' function
+        """
+        # mock request and raise error
+        mocked_request.side_effect = requests.Timeout
+
+        # set config
+        config = {
+            "access_token": "test_access_token",
+            "user_agent": "test_user_agent"
+        }
+
+        # verify that we raise Timeout error when using "with" statement
+        with self.assertRaises(requests.Timeout):
+            # create client and call function
+            client = TikTokClient(config.get("access_token"), config.get("user_agent"), config.get("request_timeout"))
+            client.request("GET")
+
+        # get arguments passed during calling "requests.Session.request"
+        args, kwargs = mocked_request.call_args
+
+        # get url from the args and verify the URL is called for "check_access_token" function
+        self.assertRegex(args[1], r"/user/info")
+        # verify that we backoff for 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+
+    @mock.patch("tap_tiktok_ads.client.TikTokClient.check_access_token")
+    def test_request(self, mocked_check_access_token, mocked_request, mocked_sleep):
+        """
+            Verify we backoff for Timeout error when calling API from 'request' function
+        """
+        # mock request and raise error
+        mocked_request.side_effect = requests.Timeout
+
+        # set config
+        config = {
+            "access_token": "test_access_token",
+            "user_agent": "test_user_agent"
+        }
+
+        # verify that we raise Timeout error when using "with" statement
+        with self.assertRaises(requests.Timeout):
+            # create client and call function
+            client = TikTokClient(config.get("access_token"), config.get("user_agent"), config.get("request_timeout"))
```

Comment:
Move this client initialization out of the `with` statement, so it makes sure that the error is raised from the client.request.
Also, make same changes in other functions wherever it applies

Reply:
Updated.